tools/operator/scripts/validate.sh

```diff
@@ -0,0 +1,146 @@
+#!/bin/bash -e
+
+if [[ $# != 6 && $# != 9 ]]; then
+  echo "Invalid number of arguments. Usage:"
+  echo "./validate.sh <kubeconfig path> <MSPID> <artifactsLocation> <ordererOrganizations list> <numOrderer in each org> <numChannels> <peerOrganizations list> <numpeers in each org> <peerOrganizations MSPID list>"
+  exit 1
+fi
+
+export KUBECONFIG=$1
+export CORE_PEER_LOCALMSPID=$2
+ARTIFACTS_LOCATION=$3
+IFS=,
+ORDERER_ORGS=($4)
+NUM_ORDERERS=($5)
+PEER_ORGS=($7)
+NUM_PEERS=($8)
+PEER_ORGS_MSPID=($9)
+unset IFS
+NUM_CHANNELS=$6
+ORDERERS=()
+CHANNELS=("orderersystemchannel")
+PORT=30000
+FLAG=0
```

Comment:
Instead of defining it in each and every function where I need it, I declared it as a global variable and used it in the functions that need this variable Scott

Reply:
How about we delete the FLAG and just use LATEST_BLOCK? Set it to zero here, and in other appropriate places (where you currently set FLAG to zero). A nonzero value means you have obtained the LATEST_BLOCK_FOR_CHANNEL

Reply:
I used FLAG just for one purpose. Initially, we don't know the lastest block height. So, if the FLAG value is 0, then I will fetch the latest config block from an orderer or a peer and will change the FLAG value. If we are fetching the latest config block for another orderer, this time the FLAG value is non zero and so it will directly compare with the block number that got stored when the FLAG value is zero. That's what I thought of and implemented this Scott.

Reply:
For code maintenance, it is better to define your variables with as much local scope as possible, and use more specific names. If not, then as functions evolve and get called from different places, then you may encounter problems with reusing a global variable for different purposes.

Reply:
Could you please elaborate on this one @scottz64 . Are you saying regarding only the FLAG variable or all the variables? If all the variables, could you give me an example?